tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenStore.scala

```diff
@@ -0,0 +1,111 @@
+package com.linagora.tmail.james.jmap.longlivedtoken
+
+import com.google.common.base.Preconditions
+import org.apache.commons.lang3.tuple.ImmutableTriple
+import org.apache.james.core.Username
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.util.UUID
+import scala.collection.concurrent.Map
+import scala.util.Try
+
+object LongLivedTokenId {
+  def parse(string: String): Either[IllegalArgumentException, LongLivedTokenId] = Try(UUID.fromString(string))
+    .map(LongLivedTokenId(_))
+    .fold(e => Left(new IllegalArgumentException("LongLivedTokenId must be backed by a UUID", e)), Right(_))
+
+  def generate: LongLivedTokenId = LongLivedTokenId(UUID.randomUUID())
+}
+
+object LongLivedTokenSecret {
+  def parse(string: String): Either[IllegalArgumentException, LongLivedTokenSecret] = Try(UUID.fromString(string))
+    .map(LongLivedTokenSecret(_))
+    .fold(e => Left(new IllegalArgumentException("LongLivedTokenSecret must be backed by a UUID", e)), Right(_))
+
+  def generate: LongLivedTokenSecret = LongLivedTokenSecret(UUID.randomUUID())
+}
+
+case class DeviceId(value: String) extends AnyVal
+
+case class LongLivedTokenId(value: UUID)
+
+case class LongLivedTokenSecret(value: UUID)
+
+case class LongLivedToken(deviceId: DeviceId, secret: LongLivedTokenSecret)
+
+case class LongLivedTokenFootPrint(id: LongLivedTokenId, deviceId: DeviceId)
+
+case class LongLivedTokenNotFoundException() extends RuntimeException
+
+trait LongLivedTokenStore {
+  def store(username: Username, longLivedToken: LongLivedToken): Publisher[LongLivedTokenId]
+
+  def validate(username: Username, secret: LongLivedTokenSecret): Publisher[LongLivedTokenFootPrint]
+
+  def listTokens(user: Username): Publisher[LongLivedTokenFootPrint]
+
+  def revoke(username: Username, id: LongLivedTokenFootPrint): Publisher[Unit]
+}
+
+class InMemoryLongLivedTokenStore(tokenStore: Map[Username, Seq[ImmutableTriple[DeviceId, LongLivedTokenId, LongLivedTokenSecret]]] = scala.collection.concurrent.TrieMap())
```

Comment:
You should initialise the map within the class as a `val` and not leak the underlying map in the constructor to the outer world.